lib/libvfio-user.c

```diff
@@ -140,36 +140,31 @@ dev_get_caps(vfu_ctx_t *vfu_ctx, vfu_reg_info_t *vfu_reg, bool is_migr_reg,
             sparse = (struct vfio_region_info_cap_sparse_mmap*)header;
         }
 
-        if (nr_mmap_areas > vfu_ctx->client_max_fds) {
-            vfu_log(vfu_ctx, LOG_DEBUG, "%s: region has nr_mmap_areas=%d, "
-                    "but client only supports %d fds", __func__,
-                    nr_mmap_areas, vfu_ctx->client_max_fds);
-            return ERROR_INT(ENOSPC);
-        }
-
         *fds = malloc(nr_mmap_areas * sizeof(int));
         if (*fds == NULL) {
             return ERROR_INT(ENOMEM);
         }
         sparse->header.id = VFIO_REGION_INFO_CAP_SPARSE_MMAP;
         sparse->header.version = 1;
         sparse->header.next = 0;
-        sparse->nr_areas = *nr_fds = nr_mmap_areas;
+        sparse->nr_areas = nr_mmap_areas;
```

Comment:
split out this fix into a separate PR?

Reply:
also could add a test maybe?

Reply:
> split out this fix into a separate PR?

yes

> also could add a test maybe?

We already have `test_device_get_region_info_caps()`, I'll extend it.